### PR TITLE
Add collapsible debug panel display agent info (#547)

### PR DESCRIPTION
**Pull request by JingfeiPeng**
Added a left control panel on envision to select what information to display. 

https://user-images.githubusercontent.com/36685292/107818087-0e003e00-6d45-11eb-8438-da764e5e6814.mp4

**Summary of changes:**

**SMARTS side:**
Added `ego_agent_ids` `position` `speed` `heading` `lane_ids` fields in the state definition we are sending to envision from smarts.py. Collected these data in smarts.py `_try_emit_envision_state`

**Envision side:**
Added a `ControlPanel` component which is the control panel on the left side of envision.
Added `controlModes` in app.js as flags to show which attributes to display. `ControlPanel` calls `toggleControlModes` in app.js to change `controlModes` in app.js
Refactored the original `AgentScores` component to be a generic `InfoDisplay` component to display all attribute data in table form.
Added a `div` in `simulation.js` to stack all the InfoDisplay components and display each conditionally based on `controlModes`.

**Remaining concerns:**
I am also not sure if I need to add the control panel in SimulationGroup component. There is a SimulationGroup component in app.js under "/all" route. But when I go to "http://localhost:8081/all" it shows page not found. I am not sure how to get to that route.

When all the 5 attributes were selected, there are lots data on the page, but I guess this isn't a problem as user can just select the attributes they want.

### CODE REVIEW

**Review by Gamenot — APPROVED** [review]

File: smarts/core/smarts.py [hunk 1]
```diff
@@ -923,6 +927,14 @@ def _try_emit_envision_state(self, provider_state, obs, scores):
                     driven_path=driven_path,
                     mission_route_geometry=mission_route_geometry,
                 )
+                speed[agent_id] = v.speed
+                position[agent_id] = v.pose.position[:2]
+                heading[agent_id] = v.pose.heading
+                if (
+                    len(vehicle_obs.waypoint_paths) > 0
+                    and len(vehicle_obs.waypoint_paths[0]) > 0
+                ):
```
[the remark this answers is not in the document]
Comment (JingfeiPeng) [reply smarts/core/smarts.py:936]: I haven't encounter any cases when this is not true, This is only not true if vehicle has no way points.
When this is not true, the default on js side is an empty dict which is ok.

**Review by Gamenot — CHANGES_REQUESTED** [review]
Testing on my side I found that the toggle works for turning on but not off. Could you confirm it works as expected on your side?

**Comment by JingfeiPeng** [comment]
> Testing on my side I found that the toggle works for turning on but not off. Could you confirm it works as expected on your side?
Right... I also met this issue sometimes. When I was testing the debug panel on the development version it always works for on and off. Then if I use the build version it has a high chance that the panel doesn't work for turning off  on the first time the page loads. But if I refresh the page and check again it works again for turning on and off.

**Comment by JingfeiPeng** [comment]
There was an error when toggling the control using checkbox (the other way is to click on the text). Fixed this now.

**Review by Gamenot — APPROVED** [review]
LGTM :shipit:

**Comment by Adaickalavan** [comment]
In my testing, the sidebar does appear in Mozilla Firefox browser, but does not appear in Google Chrome browser.

**Comment by JingfeiPeng** [comment]
@Adaickalavan I see. I think it is because Chrome caches website files, if you see "Show Scores" on the top right besides "egocentric view" then it is probably the case. [https://www.google.com/search?q=chrome+clear+website+cache&oq=chrome+clear+web&aqs=chrome.0.0j69i57j0l2j0i22i30l4.7362j0j7&sourceid=chrome&ie=UTF-8](url) this should clear old the website files. I just tested on chrome the debug panel showed up.

**Review by Adaickalavan — APPROVED** [review]

**Pull request merged**